### PR TITLE
Feature/article confirmation

### PR DESCRIPTION
**Pull request by nburkley**
Adds slack_message_controller to handle interactive message buttons for article confirmation.
aug-22-2016 16-39-28

### CODE REVIEW

**Comment by nburkley** [comment]
ping @edenspiekermann/backend

File: test/controllers/slack_message_controller_test.exs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  test "saves article and returns a message for a 'yes' confirm_article action", %{conn: conn, user: user} do
+    payload = Poison.encode!(
+      %{
+        token: slack_token,
+        user: %{ id: user.slack_id},
+        callback_id: "confirm_article",
+        actions: [ %{name: "yes", value: "http://www.example.com"} ]
+      }
+    )
+    conn = post conn, slack_message_path(conn, :new), %{ payload: payload}
```
Comment (ghost) [inline test/controllers/slack_message_controller_test.exs]: Whitespace after curly brackets `%{ payload: payload}`.

File: test/controllers/team_controller_test.exs [hunk 2]
```diff
@@ -2,8 +2,13 @@ defmodule EspiDni.TeamControllerTest do
   use EspiDni.ConnCase
 
   alias EspiDni.Team
-  @valid_attrs %{slack_id: "some content", token: "some content", url: "some content"}
-  @invalid_attrs %{}
+  @valid_attrs %{slack_id: "some content", token: "some content", name: "teamname", url: "some content"}
+  @invalid_attrs %{ token: "", slack_id: "", name: "" }
```
Comment (ghost) [inline test/controllers/team_controller_test.exs]: To much whitespace in the Map.

File: test/models/article_test.exs [hunk 3]
```diff
@@ -15,4 +15,9 @@ defmodule EspiDni.ArticleTest do
     changeset = Article.changeset(%Article{}, @invalid_attrs)
     refute changeset.valid?
   end
+
+  test "changeset with url missing" do
```
Comment (ghost) [inline test/models/article_test.exs]: Should it rather say sth.like: "changeset with url missing is invalid" to make it a bit more clear/readable?

File: web/models/article.ex [hunk 4]
```diff
@@ -20,14 +20,14 @@ defmodule EspiDni.Article do
...
-    |> validate_url(:url, message: "#{params.url} is not a valid URL!")
+    |> validate_url(:url)
   end
 
   def validate_url(changeset, field, options \\ []) do
     validate_change changeset, field, fn _, url ->
       case url |> String.to_char_list |> :http_uri.parse do
         {:ok, _} -> []
-        {:error, msg} -> [{field, options[:message] || "invalid url: #{inspect msg}"}]
+        {:error, msg} -> [{field, options[:message] || "#{url} is not a valid URL!"}]
```
Comment (ghost) [inline web/models/article.ex]: Should it use a line break  after the `->` ? Might make it a bit more readable.

File: test/controllers/slack_message_controller_test.exs (hunk 1 above)
Comment (nburkley) [inline test/controllers/slack_message_controller_test.exs]: yup, will fix

File: test/controllers/team_controller_test.exs (hunk 2 above)
Comment (nburkley) [inline test/controllers/team_controller_test.exs]: yup, will fix

File: test/models/article_test.exs (hunk 3 above)
Comment (nburkley) [inline test/models/article_test.exs]: yeah, makes more sense

File: web/models/article.ex (hunk 4 above)
Comment (nburkley) [inline web/models/article.ex]: yeah, will do.

**Pull request merged**